org.lflang/src/org/lflang/generator/ts/TSReactionGenerator.kt

```diff
@@ -349,6 +355,13 @@ class TSReactionGenerator(
         if (reactor.isFederated) {
             generatedReactions = LinkedList<Reaction>()
             for (reaction in reactor.reactions) {
+                // TODO(hokeun): Find a better way to gracefully handle this skipping.
+                // Do not add reactions created by generateNetworkOutputControlReactionBody
```

Comment:
Can't the body of the control reactions just be empty for now, instead of skipping them altogether?

Reply:
Actually, I think it's cleaner to skip these reactions until we figure out what is the right thing to do, I mean in terms of implementation. We have to handle the special case of generating an empty body for these reactions and I'm concerned that will make the code much more complicated. Please let me know if my concern makes sense.

Reply:
I think this string matching that is happening here is more prone to breakage than just generating an empty reaction. What if the name of the reaction changed? Are you worried about cluttering the generated code?

Reply:
I'm not sure if I'm following "what if the name of the reaction changed". This code doesn't depend on the name of the reaction.

It just uses the string created inside the overridden function, generateNetworkOutputControlReactionBody(). So as long as the TypeScript-specific code generation from generateNetworkOutputControlReactionBody() doesn't change, we're fine. 

What I'm worried about is that I may have to have temporary codes in multiple places (i.e., in multiple .kt) and it can become harder to maintain later.

May I ask what benefits we can get by having an empty body rather than skipping the reaction entirely?

Reply:
Ah, by the way, I'm not sure what generateNetworkOutputControlReactionBody() does. Would this be necessary for all targets? Or it may not be used at all by some targets?

Reply:
These reactions are essential to enable support for distributed cycles. You can sometimes receive PTAGs from the RTI and need these reactions in place to block certain reactions from execution using the input control reactions until you either receive a follow-up TAG or a port absent (via an output control reaction), which in both cases will unblock the input control reactions.

Reply:
Got it. Thank you for your explanation! I think we will need to implement that functionality in reactor-ts first before supporting these reactions here.